enhancements/ingress/externaldns-operator.md

```diff
@@ -0,0 +1,261 @@
+- [Release Signoff Checklist](#release-signoff-checklist)
+- [Summary](#summary)
+- [Motivation](#motivation)
+  - [Goals](#goals)
+  - [Non-Goals](#non-goals)
+- [Proposal](#proposal)
+  - [User Stories](#user-stories)
+  - [Implementation Details](#implementation-details)
+  - [Risks and Mitigations](#risks-and-mitigations)
+- [Design Details](#design-details)
+  - [Open Questions](#open-questions)
+  - [Test Plan](#test-plan)
+  - [Graduation Criteria](#graduation-criteria)
+  - [Upgrade and Downgrade Strategy](#upgradedowngrade-strategy)
+  - [Version Skew Strategy](#version-skew-strategy)
+- [Implementation History](#implementation-history)
+- [Drawbacks](#drawbacks)
+- [Alternatives](#alternatives)
+- [Infrastructure Needed](#infrastructure-needed)
+<!-- /toc -->
+
+## Release Signoff Checklist
+
+- [X] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+This enhancement proposal is for adding an operator to manage
+[ExternalDNS](https://github.com/kubernetes-sigs/external-dns). ExternalDNS has been chosen for managing external DNS
+requirements of OpenShift clusters. Initially, the operator will focus on managing external DNS record(s) of OpenShift
+[Routes](https://github.com/openshift/api/blob/master/route/v1/types.go). The operator will have its own CRD that will
+initially expose limited configuration capabilities. Existing operators will be used to guide the design of the
+ExternalDNS Operator. Tooling will be included to make it easy to build, run, test, etc. the operator for OpenShift.
+
+## Motivation
+
+The [operator pattern](https://kubernetes.io/docs/concepts/extend-kubernetes/operator/) is generally accepted, and the
+benefits of operators to OpenShift are well-known.  This enhancement proposal aims to bring the benefits of operators to
+ExternalDNS. ExternalDNS is a controller that will manage external DNS records. As with other platform components,
+ExternalDNS requires an operator to provide lifecycle management of the component.
+
+### Goals
+
+* Explore the use of an operator for managing ExternalDNS.
+* Create patterns, libraries and tooling so that ExternalDNS is of high quality, consistent in its API
+surface (common fields on CRDs, consistent labeling of created resources, etc.), yet is easy to build.
+* Build an operator that is suitable for production use of ExternalDNS.
+* Support the [CRD source](https://github.com/kubernetes-sigs/external-dns/tree/master/docs/contributing/crd-source) and
+ExternalDNS [providers](https://github.com/kubernetes-sigs/external-dns/tree/master/provider) relevant to OpenShift.
+  
+### Non-Goals
+
+* Replace the functionality of existing operators. __Note:__ The ExternalDNS Operator intends to replace external DNS
```

Comment:
Could you be a bit more specific here? If the ExternalDNS Operator will later down the road replace the external DNS features provided by the Ingress Operator, than what existing functionality won't be replaced? Perhaps replacing the external DNS management available via the ingress operator should be listed as a goal here to avoid confusion?